packages/components/src/presentation-components/table/UsePresentationTable.ts

```diff
@@ -84,10 +96,59 @@ export function usePresentationTable<TColumn, TRow>(props: UsePresentationTableP
  */
 export function usePresentationTableWithUnifiedSelection<TColumn, TRow>(
   props: Omit<UsePresentationTableProps<TColumn, TRow>, "keys">,
-): UsePresentationTableResult<TColumn, TRow> {
+): UsePresentationTableWithUnifiedSelectionResult<TColumn, TRow> {
   const unifiedSelection = useUnifiedSelectionContext();
   const keys = unifiedSelection?.getSelection() ?? emptyKeySet;
-  return usePresentationTable({ ...props, keys });
-}
+  const [selectedRows, setSelectedRows] = useState<TableRowDefinition[]>();
+
+  const { imodel, ruleset, pageSize, columnMapper, rowMapper } = props;
+  const columns = useColumns({ imodel, ruleset, keys });
+  const { options, sort, filter } = useTableOptions({ columns });
+  const { rows, isLoading, loadMoreRows } = useRows({ imodel, ruleset, keys, pageSize, options });
+
+  const tableGuid: string = useMemo(() => window.crypto.randomUUID(), []);
```

Comment:
You can use `Guid` from `@itwin/core-bentley` to generate random guid.

```suggestion
  const tableGuid = useMemo(() => Guid.createValue(), []);
```

Reply:
Removed the creation of tableGuid as it isn't needed in the updated code.